### PR TITLE
When a build artifact version is null, show that the build is pending

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
this doesn't cover any error cases, but we've yet to have a way to capture errors, except maybe through a BuildEngineBuild record -- would need to investigate how that'd be accessed -- but that's out of scope for this PR

https://developertown.visualstudio.com/SIL%20International%20Scriptoria/_workitems/edit/34286

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/__tests__/product-files-acceptance-test.tsx [hunk 1]
```diff
@@ -11,6 +11,161 @@ import {
 
 import page from './page';
 
+async function visitTheFilesTab() {
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/__tests__/product-files-acceptance-test.tsx]: this stuff here is mostly an extraction refactor

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/index.tsx [hunk 2]
```diff
@@ -17,17 +17,17 @@ export default function Builds({ product }) {
     productBuilds: (q) => q.findRelatedRecords(product, 'productBuilds'),
   });
 
-  const [activeVersion, setActiveVersion] = useState((productBuilds || [])[0]);
-
-  const sortedBuilds = productBuilds.sort(compareVia((build) => attributesFor(build).version));
+  const sortedBuilds = productBuilds.sort(
+    compareVia((build) => attributesFor(build).version, true)
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/index.tsx]: this just inverts the sort

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/index.tsx [hunk 3]
```diff
@@ -17,17 +17,17 @@ export default function Builds({ product }) {
...
+  const [activeVersion, setActiveVersion] = useState((sortedBuilds || [])[0]);
 
   return (
     <div data-test-build>
       <ResourceSelect
         items={sortedBuilds}
         labelField={(build: ProductBuildResource) => {
-          console.log(build);
-          const version = attributesFor(build).version;
+          const version = attributesFor(build).version || ` ${t('projects.buildPending')} `;
```
Comment (abillingsley) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/index.tsx]: is the whitespace before and after `${t('projects.buildPending')}` intentional?

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/index.tsx]: it is yeah, the rendered output either be:

`Current Build (v1.0.0)`
or
`Current Build ( Build Pending )`

without the spaces it does:
`Current Build (build Pending)`
so, I guess, I should add "this" comment as an inline-comment, and then make an issue to investigate further (doing that now)

**Pull request merged**